pyspecdata/axis_manipulation/register_axis.py

```diff
@@ -1,36 +1,41 @@
-from ..general_functions import *
 import numpy as np
-if not inside_sphinx():
-    from pylab import r_
 
-def register_axis(self,arg,nearest=None):
-    r'''Interpolate the data so that the given axes are in register with a set of specified values. Does not change the spacing of the axis labels.
-    
-    It finds the axis label position that is closest to the values given in `arg`, then interpolates (Fourier/sinc method) the data onto a new, slightly shifted, axis that passes exactly through the value given.
-    To do this, it uses 
-    :func:`.ft_clear_startpoints() <pyspecdata.core.nddata.ft_clear_startpoints>`
+
+def register_axis(self, arg, nearest=None):
+    r"""Interpolate the data so that the given axes are in register with a set
+    of specified values. Does not change the spacing of the axis labels.
+
+    It finds the axis label position that is closest to the values given in
+    `arg`, then interpolates (Fourier/sinc method) the data onto a new,
+    slightly shifted, axis that passes exactly through the value given.
+    To do this, it uses
+    :func:`.ft_new_startpoint() <pyspecdata.core.nddata.ft_new_startpoint>`
     and uses
     :func:`.set_ft_prop() <pyspecdata.core.nddata.set_ft_prop>`
     to override the "not aliased" flag.
 
     Parameters
     ----------
     arg : dict (key,value = str,double)
-        A list of the dimensions that you want to place in register, and the values you want them registered to.
+        A list of the dimensions that you want to place in register, and the
+        values you want them registered to.
     nearest: bool, optional
-        Passed through to ft_clear_startpoints
-    '''
-    for k,v in arg.items():
+        Passed through to ft_new_startpoint
+    """
+    for k, v in arg.items():
         x = self.getaxis(k)
         idx = np.argmin(abs(x - v))
-        offset = (v # where I want to be
-                - x[idx]) # where I actually am
-        offset += x[0] # since the following takes the startpoint
+        offset = v - x[idx]  # where I want to be  # where I actually am
+        offset += x[0]  # since the following takes the startpoint
         if self.get_ft_prop(k):
-            self.ift(k).ft_clear_startpoints(k,t='current',f=offset,nearest=False)
-            self.set_ft_prop(k,'freq_not_aliased').ft(k)
+            self.ift(k).ft_new_startpoint(k, "f", value=offset, nearest=False)
```

Comment:
github comment to explain what this is doing?

Reply:
resets the startpoint of the axis to the offset (calculated above) and then sets the ft prop to say the frequency axis is NOT aliased (since we just reset the startpoint